sui-execution/latest/sui-adapter/src/execution_engine.rs

```diff
@@ -105,6 +105,21 @@ mod checked {
         TransactionEffects,
         Result<Mode::ExecutionResults, ExecutionError>,
     ) {
+        let shared_object_refs = input_objects.filter_shared_objects();
+        let mut transaction_dependencies = input_objects.transaction_dependencies();
+        let mut temporary_store = if Mode::is_dev_dry_run_mode() {
```

Comment:
nit: (if we need this at all) can we change the name of this configuration to reflect what it controls, rather than what that control is used for -- I struggle to come up with a good name for this, but that might be because we don't need it (see comment on next line).

Reply:
I really really do not like this sort of introspection. The mode settings should be concrete and declarative in what they are doing. This does not feel that way